src/main/java/org/myrobotlab/framework/ServiceType.java

```diff
@@ -96,181 +115,232 @@ public ServiceType(String name) {
+    } else {
+      peers.put(sr.key, sr);
+    }
+  }
+  */
+
+  public void setAvailable(boolean b) {
+    this.available = b;
   }
 
   public void setCloudService(boolean b) {
     isCloudService = b;
   }
 
-  public void setRequiresKeys(boolean b) {
-    requiresKeys = b;
+  public void setLicenseApache() {
+    addLicense("apache");
   }
 
-  public boolean requiresKeys() {
-    return requiresKeys;
+  public void setLicenseGplV3() {
+    addLicense("gplv3");
   }
 
-  public void addDependency(String groupId, String artifactId) {
-    addDependency(groupId, artifactId, null, null);
+  public void setLicenseProprietary() {
+    addLicense("proprietary");
   }
 
-  public void addDependency(String groupId, String artifactId, String version) {
-    addDependency(groupId, artifactId, version, null);
+  public void setLink(String link) {
+    this.link = link;
   }
 
-  public void addDependency(String groupId, String artifactId, String version, String ext) {
-    ServiceDependency library = new ServiceDependency(groupId, artifactId, version, ext);
-    lastDependency = library;
-    dependencies.add(library);
+  public void setRequiresKeys(boolean b) {
+    requiresKeys = b;
   }
 
-  public void exclude(String groupId, String artifactId) {
-    // get last dependency
-    // dependencies
-    if (lastDependency == null) {
-      log.error("DEPENDENCY NOT DEFINED - CANNOT EXCLUDE");
-    }
-    lastDependency.add(new ServiceExclude(groupId, artifactId));
+  public void setSponsor(String sponsor) {
+    this.sponsor = sponsor;
   }
 
-  public void includeServiceInOneJar(Boolean b) {
-    includeServiceInOneJar = b;
+  /**
+   * sharing means sharePeer is forced - while addPeer will check before adding
+   * 
+   * @param key
+   *          k
+   * @param actualName
+   *          n
+   * @param peerType
+   *          n
+   * @param comment
+   *          comment
+   */
+  public void sharePeer(String key, String actualName, String peerType, String comment) {
+    peers.put(key, new ServiceReservation(key, actualName, peerType, comment));
   }
 
-  public boolean includeServiceInOneJar() {
-    return includeServiceInOneJar;
+  public int size() {
+    return dependencies.size();
   }
 
-  public void addArtifact(String orgId, String classifierId) {
-    lastDependency.add(new ServiceArtifact(orgId, classifierId));  
+ 
+  public void setServiceName(String serviceName) {
+    this.serviceName = serviceName;
+  }
+  
+  @Override
+  public String toString() {
+    StringBuffer sb = new StringBuffer();
+    if (serviceName != null) {
+      sb.append(String.format("\n%s %s\n", serviceName, simpleName));
+    } else {
+      sb.append(String.format("\n%s\n", simpleName));
+    }
+    
+    for (ServiceReservation sr : peers.values()) {
+      sb.append(sr).append("\n");
+    }
+    
+    return sb.toString();
+  }
+
+  // GAH ! - more convertions for smaller pr :(

```

Comment:
When I ask for smaller PRs i mean a smaller number of things being fixed in the PR.  not a smaller number of lines.

Reply:
I said that making ServiceType and MetaData the same thing would touch a large number of files.  To keep the number diffs between files small (down to 35 vs 365) these converters were necessary.

Reply:
I was referring to the motor/speech and joystick changes that leaked into the other pull request.  i don't mind a boiler plate change to a lot of files. that's fine.

Reply:
sure .. but I'll save the many boiler plate updates for next pr